tools/analysis/programs.go

```diff
@@ -145,6 +176,14 @@ func (programs Programs) check(
 						return nil, err
 					}
 
+					// If the imported program has a load error, return it
+					// This may happen if a program is both imported and the entry point
+					// and has an error that was handled by a custom error handler
+					// However, we still want this error in the import resolution
+					if programs[importedLocation].loadError != nil {
+						return nil, programs[importedLocation].loadError
```

Comment:
Refactor the common expression, that is tested and then used, into a variable:

```suggestion
					loadError := programs[importedLocation].loadError
					if loadError != nil {
						return nil, loadError
```